lms/djangoapps/experiments/utils.py

```diff
@@ -362,7 +363,6 @@ def get_program_context(course, user_enrollments, audit_enrollments):
                 # program has 3 courses (A, B and C), and the user previously purchased a certificate for A.
                 # The user is enrolled in audit mode for B. The "left to purchase price" should be the price of
                 # B+C.
-                non_audit_enrollments = [en for en in user_enrollments if en not in audit_enrollments]
                 courses_left_to_purchase = get_unenrolled_courses(courses, non_audit_enrollments)
```

Comment:
Do we want this to use the new entitlements check?

Reply:
I'm not sure how that would work, honestly. The entitlements get a bit complicated to try and figure out which courses they would apply to.